i_datenbank_verbessern

schmitzdominic opened the pull request:
Folgendes wurde behandelt.

- Tables werden beim ersten Start created
- Es wurden im DBHandler diverse Funktionen implementiert (add, remove, update, get) der (C_Laden, C_Bon und C_Artikel)
- Es wurde ein AssetHelper implementiert der die Produkt DB einbindet
- Nun ist es auch möglich in der manuelle View einen neuen Laden hinzuzufügen
- Die BonListe im Bons Tab ließt nun alle Bons aus der DB.

schmitzdominic committed:
Table Klassen hinzugefügt.
- C_Artikel
- C_Bon
- C_Laden

schmitzdominic committed:
Datenbank erzeugt.
- checkTables überarbeitet.

schmitzdominic committed:
Database Handler erweitert
- get und set Funktionen für alle Tables erstellt.

schmitzdominic committed:
Bugfix mit laden id

schmitzdominic committed:
Bugfix mit diversen Überarbeitungen.

schmitzdominic committed:
Beispiele hinzugefügt.

schmitzdominic committed:
Weitere Methoden hinzugefügt
- Remove Methoden
- getBon, getArtikel, getLaden

schmitzdominic committed:
Beispiele hinzugefügt.

schmitzdominic committed:
Laden kann jetzt auch über den spinner angelegt werden

- Laden kann jetzt über den spinner angelegt werden
- Dummy Daten können nun über eine Funktion in der Main generiert werden.

schmitzdominic committed:
Artikel überarbeitet

- Artikel wurde die Kategorie hinzugefügt
- Preis des Artikels wurde in einen float geändert

schmitzdominic committed:
Produkt db hinzugefügt.

schmitzdominic committed:
Update Funktionen hinzugefügt.

schmitzdominic committed:
Standard Laeden hinzugefügt.

schmitzdominic committed:
Laeden werden nun sortiert ausgegeben.

schmitzdominic committed:
Laeden werden nun sortiert ausgegeben.

Xantey commented on app/src/main/java/de/projektss17/bonpix/A_OCR_Manuell.java:
```diff
@@ -580,7 +596,7 @@ private double getFinalPrice(){
      * @param articles Array mit Articles
      */
-    private void fillMask(Uri imageUri, String ladenName, String anschrift, String datum, String sonstiges, C_Article[] articles){
+    private void fillMask(Uri imageUri, String ladenName, String anschrift, String datum, String sonstiges, C_Artikel[] articles){
```
C_Artikel[] articles ... Das ist wirklich die Krönung was euer Denglisch angeht 😄

Xantey commented on app/src/main/java/de/projektss17/bonpix/A_OCR_Manuell.java:
```diff
@@ -607,8 +623,8 @@ private void fillMask(Uri imageUri, String ladenName, String anschrift, String d
         if(articles != null){
-            for(C_Article article : articles){
-                this.inflateEditRow(article.getName(), article.getPrice());
+            for(C_Artikel article : articles){
```
C_Artikel article - warum dann nicht artikel? Ich würde mir wünschen wir würden uns auf Englisch einigen 😄

schmitzdominic replied:
Wäre auch ein Vorschlag. Kommentare können gern auf Deutsch sein, aber der Rest ist wirklich zum Teil schlimm.

Sippl replied:
Stimmt, die Vermischungen sind echt krass, wenn man alles so aufgelistet mal sieht. Dann legen wir halt die Amtsprache Englisch fest :D

Xantey commented on app/src/main/java/de/projektss17/bonpix/A_OCR_Manuell.java:
```diff
@@ -607,8 +623,8 @@ private void fillMask(Uri imageUri, String ladenName, String anschrift, String d
-            for(C_Article article : articles){
-                this.inflateEditRow(article.getName(), article.getPrice());
+            for(C_Artikel article : articles){
+                this.inflateEditRow(article.getName(), ""+article.getPreis());
```
article.getPreis() - hier das gleiche. Die meisten Methoden haben doch englische Namen oder? 😃

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_DatabaseHandler.java:
```diff
@@ -38,70 +40,596 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
-    public ArrayList<String> getAllLaeden(SQLiteDatabase db){
-        ArrayList<String> list = new ArrayList();
-        String query = "SELECT * FROM laeden";
+    public ArrayList<C_Artikel> getAllArtikelFromBon(SQLiteDatabase db, C_Bon bon){
```
... Ihr wisst es ja sicherlich was ich meine 😄

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_DatabaseHandler.java:
```diff
@@ -38,70 +40,596 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+     * @param db Datenbank
+     * @return HashMap mit allen Zuweißungen Integer, Integer
+     */
+    public HashMap<Integer, Integer> getAllBonArtkel(SQLiteDatabase db){
```
Hier fehlt ein "i"

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_DatabaseHandler.java:
```diff
@@ -38,70 +40,596 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+     * @param artikelPreis Artikel Preis
+     * @return true - existiert, false - existiert nicht
+     */
+    public boolean checkIfArtikelExist(SQLiteDatabase db, String artikelName, float artikelPreis){
```
Bitte in Zukunft auf die Namen achten...

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_DatabaseHandler.java:
```diff
@@ -38,70 +40,596 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+     * @return true - existiert, false - existiert nicht
+     */
+    public boolean checkIfArtikelExist(SQLiteDatabase db, String artikelName, float artikelPreis){
+        for(C_Artikel artikel : this.getAllArtikel(db)){
```
C_Artikel artikel - es geht also doch? 😄

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_Laden.java:
```diff
@@ -0,0 +1,42 @@
+ * Created by Domi on 27.04.2017.
+ */
+
+public class C_Laden {
```
Standard Comments wären in dieser Klasse noch nice

Xantey reviewed (changes requested):
All in All saubere Arbeit. Über die Namenskonventionen hatten wir ja schon beim erstellen des Codes gesprochen, bitte achtet in Zukunft darauf nicht zu krass ins Denglisch abzudriften. Mal dies und mal das zu verwenden ist nicht so vorteilhaft, lieber bleiben wir einer Linie treu. 👍

schmitzdominic committed:
Marcus seine Review umgesetzt

- Großteile ins Englische übersetzt
- Standard Kommentare zu C_Laden hinzugefügt.

schmitzdominic commented:
Hi @Xantey,
Erst mal vielen dank für deine wirklich ausführliche Review.
Hab mal das meiste ins englische gebracht. Die Klassennamen hab ich noch nicht geändert. Eventuell können wir das in einem anderen Sprint noch optimieren ;)
Hier der Commit https://github.com/schmitzdominic/ProjektSS17/pull/172/commits/045563418e1cc50ff1a31a2d36d8fa9a00fd1c04

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_Artikel.java:
```diff
@@ -7,22 +7,22 @@
-    private double preis;
-    private String name, kategorie;
+    private double price;
+    private String name, categorie;
```
categories oder category - categorie gibts nicht 😄

Xantey commented on app/src/main/java/de/projektss17/bonpix/daten/C_Bon.java:
```diff
@@ -9,139 +9,139 @@
-    public C_Bon(String Bildpfad, String Ladenname, String Anschrift, String SonstigeInfos,
-                 String Datum, String garantieEnde, boolean favorite, boolean garantie, ArrayList<C_Artikel> artikel) {
-        this(0, Bildpfad, Ladenname, Anschrift, SonstigeInfos, Datum, garantieEnde, favorite, garantie, artikel);
+    public C_Bon(String path, String shopName, String adress, String otherInformations,
```
Standardcomments zum erklären der Methoden wäre vielleicht noch nice in der Klasse C_Bon

schmitzdominic replied:
Das sind die Konstruktoren :D Wirklich für jeden einzelnen Getter und Setter der Klasse? das ist wirklich viel xD

Xantey reviewed (changes requested):
Die Variable sollte noch geändert werden. Ansonsten vielen Dank für die angenommenen Änderungen, sieht sonst alles wie gewohnt super aus. 👍

schmitzdominic committed:
Mehr Dokumentiert.

schmitzdominic commented:
@Xantey 
Alles Dokumentiert ;) https://github.com/schmitzdominic/ProjektSS17/pull/172/commits/695a3923030ce879e5b89b1fb4e0383aacb4ccbb

Xantey reviewed (approved):
Jetzt sollte alles passen, sehr nice 👍

SshaBrn reviewed (approved):
WOW, sehr gute Arbeit. Code ist sehr gut dokumentiert und selbst für mich verständlich. Mir sind soweit keine Fehler mehr aufgefallen. Marcus hat ja bereits sehr umfangreich gereviewed. Von mir also ein dickes APPROVE.
